lib/matplotlib/dates.py

```diff
@@ -1254,6 +1241,8 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
+        if not isinstance(interval, int) or interval < 1:
```

Comment:
This will fail on `internal == 1.0` which is surprising.  I think `if not interval == int(interval)` is a better test here.